projects/demo/src/app/drag-and-drop/grouping.demo.ts

```diff
@@ -58,4 +58,11 @@ export class GroupingDemo {
       this.moveItem(dragEvent.dragDataTransfer, this.files, this.droppedImages);
     }
   }
+
+  @HostListener('unloaded')
+  ngOnDestroy(): void {
+    if (FILES && FILES.length) {
+      FILES.splice(0, FILES.length);
```

Comment:
Thanks for your analysis, but we cannot accept this fix as is because it actually causes a bug in or demo app.

Since the `FILES` array is owned by the module, not the component class, this actually causes a bug in the demo app.

1. Navigate to https://484--storybook-clarity-design.netlify.app/demo/drag-and-drop/grouping.
2. Click a link for another drag and drop page (e.g. “Basic Draggable”).
3. Click the “Grouping” link.
4. Notice that there are no files to drag this time.

This problem can likely be fixed by removing the `FILE` constant array and inlining its contents in the constructor when the `files` property is initialized. This way, the `ngOnDestroy` logic can be removed entirely since the `files` class property will be dropped by garbage collection when the class instance is dropped.  

That said, it may not be worth it to test my proposed fix because this code is removed on our `beta` branch (which will be promoted `main` soon). See #426.